erpnext/erpnext_integrations/doctype/leaflink_settings/leaflink_settings.py

```diff
@@ -0,0 +1,148 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2021, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import frappe
+import json
+import pprint
+from frappe.utils import add_days
+import datetime
+import requests
+from frappe.model.document import Document
+
+class LeafLinkSettings(Document):
+	pass
+
+
+@frappe.whitelist()
+def sync_item_with_leaflink(item_details):
+	"""
+	Sync an item in leaflink by hitting the Sync with LeafLink Button.
+	"""
+	pass
+
+
+@frappe.whitelist(allow_guest=True)
+def sync_customer_with_leaflink(**customer_data):
+	"""
+	Sync a customer with leaflink by hitting Sync with Customer Button.
+	"""
+	leaflink_settings = frappe.get_cached_doc("LeafLink Settings")
+	header = {
+		leaflink_settings.get("api_key"): leaflink_settings.get("api_value")
+	}
+
+	#check if customer exists on leaflink
+	customers_url = leaflink_settings.get("customers_url")
+	response = requests.get(customers_url, headers=header, params={"name": customer_data.get("name")}).json()
+
+	#create customer record on leaflink if it does not exist
+	if not response.get("count"):
+
+		#fetch default license of customer from bloomstack
+		(license_no, license_type) = frappe.db.sql('''SELECT
+				license, license_type
+				FROM
+				`tabCompliance License Detail`
+				WHERE parent = %(customer_name)s
+				''', {"customer_name": customer_data.get("name")})[0]
+
+		#create new customer data
+		new_customer = {
+			"name": customer_data.get("name"),
+			"owner": leaflink_settings.get("leaflink_id")
+	 	}
+
+		#post request to create customer in leaflink
+		post_response = requests.post(customers_url, headers=header, data=new_customer)
+
+
+@frappe.whitelist(allow_guest=True)
```

Comment:
There should be an authentication for this endpoint. since its open to everyone.